docs/how-to-guides/routes.md

```diff
@@ -0,0 +1,216 @@
+
+### `route` – Standard route
+
+Used to define a route for a specific path.
+
+```ts
+config.registerRoute({
+  type: 'route',
+  path: '/login',
+  file: '@plone/cmsui/routes/login.tsx',
+  options: {
+    id: 'login',
+    index: true,
+  },
+});
+```
+
+### `layout` – Layout with children
+
+Useful for defining shared page layouts for a set of routes.
+
+```ts
+config.registerRoute({
+  type: 'layout',
+  file: '@my-addon/routes/layouts/DashboardLayout.tsx',
+  children: [
+    {
+      type: 'route',
+      path: 'home',
+      file: '@my-addon/routes/DashboardHome.tsx',
+    },
+    {
+      type: 'route',
+      path: 'settings',
+      file: '@my-addon/routes/DashboardSettings.tsx',
+    },
+  ],
+});
+```
+
+### `index` – Default route in a context
+
+Rendered when no sub-route is matched. Must be placed inside a `layout` or `prefix`.
+
+```ts
+config.registerRoute({
+  type: 'index',
+  file: '@my-addon/routes/HomePage.tsx',
+});
+```
+
+### `prefix` – Structural grouping without layout
+
+Groups routes under a common path prefix without introducing a layout component.
+
+```ts
+config.registerRoute({
+  type: 'prefix',
+  path: '/blog',
+  children: [
+    {
+      type: 'route',
+      path: ':id',
+      file: '@my-addon/routes/BlogPost.tsx',
+    },
+    {
+      type: 'route',
+      path: 'archive',
+      file: '@my-addon/routes/BlogArchive.tsx',
+    },
+  ],
+});
+```
+
+## Example: Defining a real Seven core add-on route
+
+Let’s look at a real-world example from the `@plone/cmsui` add-on.
+The following route registration defines the `/edit` interface in Seven.
+This example demonstrates how to nest a `prefix` route under a `layout`, and use both `index` and wildcard routes within the same branch.
+
+```ts
+config.registerRoute({
+  type: 'layout',
+  file: '@plone/cmsui/routes/layout.tsx',
+  children: [
+    {
+      type: 'prefix',
+      path: 'edit',
+      children: [
+        {
+          type: 'index',
+          file: '@plone/cmsui/routes/edit.tsx',
+          options: {
+            id: 'index-edit',
+          },
+        },
+        {
+          type: 'route',
+          path: '*',
+          file: '@plone/cmsui/routes/edit.tsx',
+        },
+      ],
+    },
+  ],
+});
+```
+
+### What this defines:
+
+- A top-level layout component from `@plone/cmsui/routes/layout.tsx`.
+- A `prefix` path `edit` that groups all `/edit/*` routes.
+- An `index` route for `/edit`.
+- A wildcard route for any `/edit/*` subpath.
+
+```{note}
+Even though both the `index` and `*` routes use the same file, you could conditionally render different content based on `useParams()` or other context inside the component.
+```
+
+---
+
+### File structure for `@plone/cmsui` edit route
+
+```
+@plone/cmsui/
+├── routes/
+│   ├── layout.tsx       ← Shared layout used as base for CMS UI views
+│   └── edit.tsx         ← Component used for /edit and all subpaths
+└── index.ts             ← Entry point where `config.registerRoute()` is called
```

Comment:
```suggestion
│   ├── layout.tsx
│   └── edit.tsx
└── index.ts
```

Reply:
I think this is useful...

Reply:
It's useful, yes, but it has shortcomings. ASCII art diagrams with narrative text work against people with visual impairments. Code blocks don't index in search engines well, and might not get translated correctly. The explanations are sentence fragments and may be difficult to interpret in other languages.

Splitting the narrative text from the diagram has none of those shortcomings.